subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r40/BuildCacheCleanupOperationsCrossVersionSpec.groovy

```diff
@@ -50,10 +52,19 @@ class BuildCacheCleanupOperationsCrossVersionSpec extends ToolingApiSpecificatio
                 description = "Generates a 1MB file"
             }
         """
+
+        def cacheLimit
+        sizeBasedCleanup = targetVersion < GradleVersion.version("4.6")
```

Comment:
I think we need to check that `targetVersion.baseVersion < GradleVersion.version("4.6")`, right? If the target version is a `4.6` release snapshot, then we don't have size based cleanup, but the snapshot version would be smaller than `4.6`.

Reply:
Good point, fixed.